settings/request_test.go

```diff
@@ -0,0 +1,690 @@
+	resp, err = conn.Do(NewSQLVDBEDebugSetRequest(true)).Get()
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, []interface{}{[]interface{}{"sql_vdbe_debug", true}}, resp.Data)
+
+	// Fetch current setting value.
+	resp, err = conn.Do(NewSQLVDBEDebugGetRequest()).Get()
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, []interface{}{[]interface{}{"sql_vdbe_debug", true}}, resp.Data)
+
+	// To test real effect we need a Tarantool instance built with
+	// `-DCMAKE_BUILD_TYPE=Debug`.
+}
+
+func TestSessionSettings(t *testing.T) {
+	skipIfSettingsUnsupported(t)
+
+	var resp *tarantool.Response
+	var err error
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	// Set some settings values.
+	resp, err = conn.Do(NewSQLDefaultEngineSetRequest("memtx")).Get()
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, []interface{}{[]interface{}{"sql_default_engine", "memtx"}}, resp.Data)
+
+	resp, err = conn.Do(NewSQLFullColumnNamesSetRequest(true)).Get()
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, []interface{}{[]interface{}{"sql_full_column_names", true}}, resp.Data)
+
+	// Fetch current settings values.
+	resp, err = conn.Do(NewSessionSettingsGetRequest()).Get()
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Subset(t, resp.Data,
+		[]interface{}{
+			[]interface{}{"sql_default_engine", "memtx"},
+			[]interface{}{"sql_full_column_names", true},
+		})
+}
+
+func TestRequestsAsync(t *testing.T) {
+	tests := []struct {
+		req   tarantool.Request
+		async bool
+	}{
+		{req: NewErrorMarshalingEnabledSetRequest(false), async: false},
+		{req: NewErrorMarshalingEnabledGetRequest(), async: false},
+		{req: NewSQLDefaultEngineSetRequest("memtx"), async: false},
+		{req: NewSQLDefaultEngineGetRequest(), async: false},
+		{req: NewSQLDeferForeignKeysSetRequest(false), async: false},
+		{req: NewSQLDeferForeignKeysGetRequest(), async: false},
+		{req: NewSQLFullColumnNamesSetRequest(false), async: false},
+		{req: NewSQLFullColumnNamesGetRequest(), async: false},
+		{req: NewSQLFullMetadataSetRequest(false), async: false},
+		{req: NewSQLFullMetadataGetRequest(), async: false},
+		{req: NewSQLParserDebugSetRequest(false), async: false},
+		{req: NewSQLParserDebugGetRequest(), async: false},
+		{req: NewSQLRecursiveTriggersSetRequest(false), async: false},
+		{req: NewSQLRecursiveTriggersGetRequest(), async: false},
+		{req: NewSQLReverseUnorderedSelectsSetRequest(false), async: false},
+		{req: NewSQLReverseUnorderedSelectsGetRequest(), async: false},
+		{req: NewSQLSelectDebugSetRequest(false), async: false},
+		{req: NewSQLSelectDebugGetRequest(), async: false},
+		{req: NewSQLVDBEDebugSetRequest(false), async: false},
+		{req: NewSQLVDBEDebugGetRequest(), async: false},
+	}
+
+	for _, test := range tests {
+		if async := test.req.Async(); async != test.async {
+			t.Errorf("An invalid async %t, expected %t", async, test.async)
+		}
+	}
+}
```

Comment:
Still need to add unit-tests for methods:

Code(), Body(), Context(), Ctx()

It would be good to split integration tests and unit tests into two files: tarantool_test.go? (integration), request_test.go (unit).

Reply:
Separated, added simple unit tests (without Body buffer comparison)

Reply:
> without Body buffer comparison

If we want complete beauty, it is better to add (SelectRequest.Body() == GetRequest.Body(), UpdateRequest.Body() == SetRequest.Body()), but it doesn't look critical, so up to you.